atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/RowResult.java

```diff
@@ -138,18 +142,29 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass()) {
             return false;
         }
-        @SuppressWarnings("rawtypes")
         RowResult other = (RowResult) obj;
-        if (columns == null) {
-            if (other.columns != null) {
+        if (columns == null && other.columns == null) {
```

Comment:
I am actually not convinced this is a good idea.

It sucks that this doesn't just work out of the box, but RowResult is generic over T and so it's defendable that it just works like everything else: relies on the values having a correct and useful #equals.

This test passes:

```
    @Test
    public void testRowResultEquality() {
        RowResult<Bytes> left =
                RowResult.of(Cell.create(new byte[] {1, 2, 3}, new byte[] {4, 5, 6}), Bytes.from(new byte[] {7, 8, 9}));
        RowResult<Bytes> right =
                RowResult.of(Cell.create(new byte[] {1, 2, 3}, new byte[] {4, 5, 6}), Bytes.from(new byte[] {7, 8, 9}));

        assertThat(left.equals(right)).isTrue();
    }
```

So it compares the keys correctly, because it uses the lexicographical comparator.